05_stringer/willshen8/stringer.go

```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+type IPAddr [4]byte
+
+func (ip IPAddr) String() string {
+	var returnIPAddress string
+	for _, v := range ip {
+		returnIPAddress += strconv.Itoa(int(v)) + "."
+	}
+	returnIPAddress = strings.TrimSuffix(returnIPAddress, ".")
```

Comment:
Given that the underlying slice is of a fixed length and of a predefined order the loop and trim could be simplified down to:
```
returnIpAddress = fmt.Sprintf("%v.%v.%v.%v", ip[0], ip[1], ip[2], ip[3])
```

Reply:
Thanks, this is a neater approach.